staging/src/k8s.io/kubectl/pkg/cmd/wait/wait_test.go

```diff
@@ -1173,7 +1173,7 @@ func TestWaitForDifferentJSONPathExpression(t *testing.T) {
 			jsonPathExp:  "{.foo.bar}",
 			jsonPathCond: "baz",
 
-			expectedErr: "foo is not found",
+			expectedErr: "timed out waiting for the condition on theresource/foo-b6699dcfb-rnv7t",
```

Comment:
I wonder is there any script out there relying on `foo is not found` error message of `wait` command.

Reply:
Probably but then we lock ourselves into supporting what is essentially a bug if we force ourselves to take that into consideration.

Reply:
To produce the same `foo is not found` error message can run the parser with `AllowMissingKeys(false)` once we reach the timeout.

Reply:
Wonder if a middle ground would be having a flag for allowing missing keys in the command that way it continues to function as it has been and adds this new ability without potentially breaking scripts downstream.

Reply:
Sorry by middle ground I just meant keep the default for the command the way it is so it doesn't potentially break anything for downstream users, and add a flag that users can use to ignore missing keys, so that we basically enable both options. It would serve to both ensure no unexpected changes for downstream users and act as a kind of guard against users potentially adding a path to a key that doesn't exist unless they specifically are ok with it being missing.

Reply:
The middle ground would only apply to jsonpath, b/c as mentioned in the original issue that is possible when waiting for conditions already, so I'm considering this more like a bug, and aligning the functionality at this time.